### PR TITLE
feat: support circleci PRs and releases

### PR DESCRIPTION
**Pull request by markowsiak**
Adding support for CircleCI.

* circleci uses the `CIRCLE_PULL_REQUEST` env var to designate the pull request URL (I pull the id from it in the code)
* circleci uses the `CIRCLE_BRANCH` env var to designate the branch
* added some description in the `READEME` on how to setup auto releasing from master builds

Testing `nlm` with circleci here in this PR: https://github.com/Automattic/vip-search-replace/pull/1
And am currently not getting auto labelling of semver updates on the PR without these changes.

Please feel free to add a `hacktoberfest-accepted` label to the PR if you feel it's worthy 👍  🤓 🙇‍♂️

### CODE REVIEW

**Comment by markowsiak** [comment]
cc: @aaarichter @aotarola  👋  Hey guys!

File: README.md [hunk 1]
```diff
@@ -83,6 +83,25 @@ build:
   <% } %>
 ```
 
+#### CircleCI
+
+You may reference teh official [circleci docs](https://circleci.com/docs/2.0/env-vars/) on setting up environment variables
```
Comment (aaarichter) [inline README.md:88]: small typo: `teh` -> `the`

**Review by aaarichter — APPROVED** [review]
Hey @markowsiak 👋  thanks for adding support.

Everything looks great but there is a tiny typo.

🚢 🇮🇹

**Comment by markowsiak** [comment]
Thanks so much for the review and catch @aaarichter 👍 
I'd love to ship it myself, but no longer have the ability.  Typo fix is up!
Hope you're doing well 😄

**Comment by aaarichter** [comment]
merged @markowsiak  - yes doing good - we had Ottercon on Tuesday and today. It was great 🥳

**Comment by markowsiak** [comment]
@aaarichter Oh man! I'd love to hear what you've all been up to.  We should set up an alumni slack or something.  Glad to hear all is well 🎉  and thanks so much for the review and merge 🦦

**Pull request merged**